### PR TITLE
Add lookup plugin to use native api

### PR DESCRIPTION
**Pull request by tanganellilore**
Hi team,

In this PR I added the functionality to use native API via lookup plugin.
 I'm simply inspired from [AWX controller_api](https://github.com/ansible/awx/blob/devel/awx_collection/plugins/lookup/controller_api.py) so the implementation it's very simple and without special checks.

Lookup working like controller_apu lookup, ex:

```
- name: List all Virtual Machines from the API
  set_fact:
    virtual_machines: "{{ lookup('ngine_io.cloudstack.api', 'listVirtualMachines') }}"

- name: List specific Virtual Machines from the API
  set_fact:
    virtual_machines: "{{ lookup('ngine_io.cloudstack.api', 'listVirtualMachines', query_params={ 'name': 'myvmname' }) }}"
```

This is very usefull for usecase like listing elements or simply query the api, without any logic that generally is present on a module (like automatic retrival of id from a name).

Let me know it this can be a good starting point.

Thanks

### CODE REVIEW

**Comment by tanganellilore** [comment]
@resmo have you see that?

**Comment by resmo** [comment]
Hi @tanganellilore 

Thanks for pinging me. I added myself as reviewer.

In general, I agree with this functionality. However, I'm in the middle of a major refactoring (see #141) that will be done by about the end of the year. Would it be ok to take a look at this PR when the dust has settled a bit?

**Comment by tanganellilore** [comment]
Hi @resmo ,

no problem, in the meantime i will use my fork with this enhancement.
let me say, is very, very simple, I create a litte wrapping to cs library, without adding logic or something similar

**Review by resmo — COMMENTED** [review]
Thanks for your suggestion. I'd like to have this lookup plugin merged for 3.0.0. However, I wonder why you created another module_utils for it. Can you explain?

**Comment by resmo** [comment]
@tanganellilore please rebase

**Comment by tanganellilore** [comment]
Hi,

to due different init and api function, moreover, to separate lookup/api module to other module.

Api lookup is very flexible without any specific check.

File: plugins/lookup/api.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# (c) 2020 Ansible Project
```
[the remark this answers is not in the document]
Comment (tanganellilore) [reply plugins/lookup/api.py:1]: will add

**Comment by resmo** [comment]
sorry again, could you please rebase again `git rebase origin/master` then force push `git push -f origin add_api_lookup` again. Thanks.

File: plugins/lookup/api.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        args = {}
+        if self.get_option("query_params"):
+            args.update(self.get_option("query_params", {}))
+
+        res = module.query_api(terms[0], **args)
+
+        if res is None:
+            return []
+        if type(res) is list:
```
Comment (resmo) [inline plugins/lookup/api.py:86]: Use isinstance() rather than type() for a typecheck.

**Pull request merged**